src/Microsoft.AspNet.Razor.Runtime/BufferedTextWriter.cs

```diff
@@ -0,0 +1,189 @@
+﻿// Copyright (c) Microsoft Open Technologies, Inc. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNet.Razor.Runtime
+{
+    /// <summary>
+    /// TextWriter which is used to write <see cref="Runtime.BufferedStringCollection"/>.
+    /// </summary>
+    public class BufferedTextWriter : TextWriter
+    {
+        private static readonly Task _completedTask = Task.FromResult(0);
+        private readonly Encoding _encoding;
+
+        /// <summary>
+        /// Creates a new instance of <see cref="BufferedTextWriter"/>.
+        /// </summary>
+        /// <param name="encoding">The character <see cref="Encoding"/> in which the output is written.</param>
+        public BufferedTextWriter(Encoding encoding)
+        {
+            _encoding = encoding;
+            BufferedStringCollection = new BufferedStringCollection();
+        }
+        
+        public BufferedStringCollection BufferedStringCollection { get; set; }
```

Comment:
could this instead be an implementation detail?  that is, expose `public IEnumerator<string> GetEnumerator()` instead